### PR TITLE
Build system: default to fmt shared library when available

### PR DESCRIPTION
**Pull request by mazes-80**
Detect system libfmt, with minimal version 5.3.
I don't know if minimal version is required, but as it is the one found in externals ...

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -530,7 +530,13 @@ if (_M_X86)
   add_subdirectory(Externals/Bochs_disasm)
 endif()
 add_subdirectory(Externals/cpp-optparse)
-add_subdirectory(Externals/fmt EXCLUDE_FROM_ALL)
+find_package(fmt 5.3) 
+if(fmt_FOUND)
+	message(STATUS "Using shared fmt ${fmt_VERSION}")
```
Comment (leoetlino) [inline CMakeLists.txt]: tab should be spaces

**Comment by orbea** [comment]
A lot of distros already provide `6.0.0`.

https://repology.org/project/fmt/versions

**Comment by mazes-80** [comment]
I updated  fmt from External, everything seems compiles fine, no breaking changes detected yet !
The switch works well as my system fmt is 5.3, build system used 6.0.0 from "static"

**Comment by leoetlino** [comment]
Hmm, it's weird that everything still builds fine because fmt/time.h was removed in 6.0, and Core (IIRC) includes it.

**Comment by BhaaLseN** [comment]
The functionality is still there, but I can't find where we use it (nor in which PR I mentioned this before).
I mean, as long as it works...

**Comment by leoetlino** [comment]
Looks like the PR is based on an old-ish commit. Please rebase and fix the build issues that will most likely appear after doing so. (I think the only change you have to make is replace fmt/time.h with fmt/chrono.h.)

**Comment by leoetlino** [comment]
Looks like fmt 6.0.0 introduces a lot of new warnings (https://github.com/fmtlib/fmt/pull/1278). Maybe we should consider disabling warnings for fmt (or even Externals in general)?

**Comment by orbea** [comment]
FWIW fmt upstream is now at `6.1.2`, this PR still being worked on?

**Comment by mazes-80** [comment]
> FWIW fmt upstream is now at `6.1.2`, this PR still being worked on?

For now using fmt 6.1.2 from system fails to compile:
Source/Core/VideoCommon/BPStructs.cpp:992:55: error: cannot bind bitfield 'left_top.X10Y10::<anonymous>.X10Y10::<unnamed struct>::x' to 'unsigned int&'
Source/Core/VideoCommon/BPStructs.cpp:1186:77: error: cannot bind bitfield 'teximg.TexImage0::<anonymous>.TexImage0::<unnamed struct>::format' to 'unsigned int&'
Source/Core/VideoCommon/BPStructs.cpp:1209:40: error: cannot bind bitfield 'teximg.TexImage1::<anonymous>.TexImage1::<unnamed struct>::tmem_even' to 'unsigned int&'
Source/Core/VideoCommon/BPStructs.cpp:1232:40: error: cannot bind bitfield 'teximg.TexImage2::<anonymous>.TexImage2::<unnamed struct>::tmem_odd' to 'unsigned int&'

Locally I just add a '+ 0' to all those complaining bitfields, it works, but I'm sure it is not the proper way to cast. What would be the good way to cast ?

**Comment by leoetlino** [comment]
Apparently this is a known issue with 6.0+. The recommended workaround is apparently to use the unary plus operator:

https://github.com/fmtlib/fmt/issues/1284#issuecomment-525778606

**Comment by mazes-80** [comment]
> Apparently this is a known issue with 6.0+. The recommended workaround is to use the unary plus operator.

Pushed fmt 6.1.2, added unary plus were they were missing.

**Comment by orbea** [comment]
@mazes-80 I don't see the new commits?

**Comment by orbea** [comment]
Fwiw this builds here.
```
$  ldd /usr/games/dolphin-emu | grep -i fmt
	libfmt.so.6 => /usr/lib64/libfmt.so.6 (0x00007feed2d98000)
```

**Pull request merged**